fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -380,111 +414,206 @@ private enum class Highlight(
-        productId: String = "123",
-        reviewGrade: ReviewQualityCheckState.Grade? = ReviewQualityCheckState.Grade.B,
-        analysisStatus: AnalysisStatus = AnalysisStatus.UP_TO_DATE,
-        adjustedRating: Float? = 3.6f,
-        productUrl: String = "",
-        highlights: SortedMap<HighlightType, List<String>>? = sortedMapOf(
-            HighlightType.QUALITY to listOf(
-                "High quality",
-                "Excellent craftsmanship",
-                "Superior materials",
-            ),
-            HighlightType.PRICE to listOf(
-                "Affordable prices",
-                "Great value for money",
-                "Discounted offers",
-            ),
-            HighlightType.SHIPPING to listOf(
-                "Fast and reliable shipping",
-                "Free shipping options",
-                "Express delivery",
-            ),
-            HighlightType.PACKAGING_AND_APPEARANCE to listOf(
-                "Elegant packaging",
-                "Attractive appearance",
-                "Beautiful design",
-            ),
-            HighlightType.COMPETITIVENESS to listOf(
-                "Competitive pricing",
-                "Strong market presence",
-                "Unbeatable deals",
-            ),
-        ),
-        recommendedProductState: ReviewQualityCheckState.RecommendedProductState =
-            ReviewQualityCheckState.RecommendedProductState.Initial,
-        productVendor: ReviewQualityCheckState.ProductVendor = ReviewQualityCheckState.ProductVendor.AMAZON,
-    ) : this(
-        productRecommendationsEnabled = productRecommendationsEnabled,
-        productAnalysis = AnalysisPresent(
-            productId = productId,
-            reviewGrade = reviewGrade,
-            analysisStatus = analysisStatus,
-            adjustedRating = adjustedRating,
-            productUrl = productUrl,
-            highlights = highlights,
-            recommendedProductState = recommendedProductState,
-        ),
-        productVendor = productVendor,
-    )
-}
+    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        ReviewQualityCheckCard(
+            modifier = Modifier
+                .fillMaxWidth()
+                .clickable(onClick = onClick),
+        ) {
+            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                Text(
+                    text = stringResource(R.string.review_quality_check_ad_title),
+                    color = FirefoxTheme.colors.textPrimary,
+                    style = FirefoxTheme.typography.headline8,
+                )
+
+                Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
+                    // todo: placeholder
+                    val imageSize = with(LocalDensity.current) { 60.dp.toPx() }.roundToInt()
+                    AsyncImage(
+                        model = ImageRequest.Builder(LocalContext.current)
+                            .data(product.imageUrl)
```

Comment:
We are fetching a resource using a network stack that is not our own. This is a hard blocker for using this library unless we can make it go through our own networking by implementing the concept-fetch API.

Reply:
https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/plugins/publicsuffixlist/build.gradle#22

?

Reply:
This is a gradle plugin, not shipped code. It's fine to use it in tooling or testing (e.g. the test eng team uses mockwebserver).

Reply:
Gotcha gotcha. For my edification then, do we need to be using our in-house networking for 100% of our (release to the wild) use cases?

Reply:
Yes, any runtime networking needs to go through Necko. We have examples in the codebase where we use our concept-fetch client around the place too.

Reply:
OK. I'll likely land this first (without the coil addition and just a placeholder ad image), and then come up with an alternative solution for image loading.

Reply:
To close this loop, Coil is no longer included in this PR